gulpfile.js/index.js

```diff
@@ -168,9 +170,54 @@ async function languagePlugins() {
 		}
 	}
 
+	/**
+	 * Inverts the given object making all values keys and all keys values.
+	 *
+	 * This essential changes the direction of all edges in the directed graph described by the given record.
+	 *
+	 * @param {Object<string, string | string[]>} rec
+	 * @returns {Object<string, string | string[]>}
+	 */
+	function invertRecord(rec) {
+		/** @type {Object<string, string[]>} */
+		const inv = {};
+		/**
+		 * @param {string} recKey
+		 * @param {string} recValue
+		 */
+		function add(recKey, recValue) {
+			(inv[recValue] = inv[recValue] || []).push(recKey);
+		}
+		for (var key in rec) {
+			var value = rec[key];
+			if (Array.isArray(value)) {
+				value.forEach(v => add(key, v));
+			} else {
+				add(key, value);
+			}
+		}
+
+		/**
+		 * This will all single-element arrays.
+		 *
+		 * @param {Object<string, string | string[]>} rec
+		 * @returns {void}
+		 */
+		function inlineElement(rec) {
+			for (const key in rec) {
+				const element = rec[key];
+				if (Array.isArray(element) && element.length === 1) {
+					rec[key] = element[0];
+				}
+			}
+		}
+		inlineElement(inv);
+		return inv;
+	}
+
 	const jsonLanguagesMap = formattedStringify(nonNullLanguageMap);
-	const jsonDependenciesMap = formattedStringify(dependenciesMap);
-	const jsonAliasMap = formattedStringify(aliasMap);
+	const jsonDependenciesMap = `invertRecord(${formattedStringify(invertRecord(dependenciesMap))})`;
+	const jsonAliasMap = `invertRecord(${formattedStringify(invertRecord(aliasMap))})`;
```

Comment:
Is this right? This looks weird. Is the intention to prefix the string with `invertRecord`?

Reply:
Autoloader has an `invertRecord` function and the prefix is there to call that function. We have to invert the structure twice to get the original back.